### PR TITLE
Deploy on Databricks

### PR DESCRIPTION
**Pull request by florian-vuillemot**
#420

### CODE REVIEW

**Review by jrobinAV — APPROVED** [review]
I believe it s good enough!
We may ask a tech writer to improve it later.

File: mkdocs.yml_template [hunk 1]
```diff
@@ -93,25 +93,31 @@ nav:
             - "Protect private files": manuals/run-deploy/run/protect_files.md
         - "Deployment":
             - manuals/run-deploy/deploy/index.md
-            - "On local machine": manuals/run-deploy/deploy/local_machine.md
-            - "On Taipy Cloud": manuals/run-deploy/deploy/taipy-cloud.md
-            - "On Linux servers":
+            - On local machine:
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]: On local machine => On the local machine

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Deploy on Databricks
+
+Databricks is a standard platform for data scientists and big data engineers. It can provide a web-based laptop connected to your data source within your cloud provider. Nevertheless, Databricks does not provide a direct path to developing a web application. This documentation will show you how to adapt your infrastructure so that you can create a Taipy application in your Databricks.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "web-based laptop"??

"...application in your Databricks"
=>
"application in your Databricks environment"

"cloud provider" -> "Cloud provider" (? not too sure... and if yes, everywhere in this document)

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+- Knowledge of Databricks.
+- Knowledge of SSH.
+- A Databricks cluster in operation.
+- A Linux-based machine that can communicate with your local machine and Databricks.
+- [:material-arrow-right: Running a Taipy application](../../run/index.md)
+
+!!! Note
+
+    If you don't have a Linux-based machine, you can use a cloud provider as Azure to
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "provider as Azure" => "provider like Microsoft Azure"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+Let's consider the following application:
+
+```python linenums="1"
+{%
+include-markdown "../../code_sample/basic_gui_rest_app.py"
+comments=false
+%}
+```
+
+Connect to Databricks and run the previous code in a Databricks Notebook. You should obtain a similar output:
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "You should obtain a similar output"
=>
"You should obtain an output similar to the following"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+## Exposing your application
+
+Here is the global architecture of the solution.
+![Global architecture](./images/global-architecture.svg)
+
+Since we can't connect to Databricks directly, we'll use an SSH tunnel between a proxy machine and the
+Databricks machine, enabling them to communicate.
+We'll also install Nginx on the Linux-based machine to route the request from your browser to your application.
+
+In the end, the browser sends a request to the Nginx on the Linux-based machine; the Nginx will redirect
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "the Nginx" => "Nginx" (twice)

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+this request to the SSH tunnel, which will redirect to the Taipy application.
+
+Here is the technical architecture of the solution.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: Final '.' => ":'"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+![Technical architecture](./images/technical-architecture.svg)
+
+!!! Warning
+
+    The Linux-based machine should be accessible from your network and Databricks.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "from your network and Databricks" => "from your network and from Databricks"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+### 1- Configure the proxy using Nginx
+
+We'll install and configure Nginx on the Linux-based machine to enable browser-based communication.
+Once configured, Nginx will redirect all incoming requests on the port HTTP (80) to port 8080.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "on the port HTTP (80)" => "from the default HTTP port (80)"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+First, install [Nginx](https://www.nginx.com/resources/wiki/start/topics/tutorials/install/). Then configure
+the redirection by putting the following content in `/etc/nginx/sites-enabled/default` :
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "by putting the following" => "by copying the following"

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        proxy_request_buffering     off;
+    }
+}
+```
+
+And restart Nginx: `systemctl restart nginx`.
+
+### 2 - Connect Databricks to the proxy machine
+
+Open Databricks on a first tab, and go to compute and select your cluster.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "on a first tab" => "in a first tab"

"and go to compute" => "press "Compute"" (?)

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+```ssh -R 8080:127.0.0.1:5000 <username>@<machine-ip>```
+
+!!! Note
+
+    If your SSH authentication is based on certificates, remember to upload them on Databricks.
+
+You are now on the Linux-based machine, and the communication is verified. You may notice the option
+`-R 8080:127.0.0.1:5000`. This option starts
+[port forwarding](https://www.ssh.com/academy/ssh/tunneling-example) from the machine to Databricks.
+Specifically, it forwards all packets from port 8080 to port 5000 on your Databricks.
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "your Databricks" => "your Databricks environment" (?)

File: docs/manuals/run-deploy/deploy/databricks/index.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+Therefore, running `curl localhost:8080` should get your application's output still running on your Databricks notebook!
```
Comment (FabienLelaquais) [inline docs/manuals/run-deploy/deploy/databricks/index.md]: "Therefore, running" => "Running"
"still" => "while"

File: mkdocs.yml_template (hunk 1 above)
Comment (FabienLelaquais) [inline mkdocs.yml_template]: Please bring back the double quotes here! Like it is done in the entirety of this template.
We may need to parse this file
Thx

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
👍

File: docs/manuals/run-deploy/deploy/databricks/index.md (hunk 11 above)
Comment (florian-vuillemot) [reply docs/manuals/run-deploy/deploy/databricks/index.md]: The "Databricks environment notebook" is a bit strange for me.

Comment (FabienLelaquais) [reply docs/manuals/run-deploy/deploy/databricks/index.md]: Indeed.
But "it forwards all packets from port 8080 to port 5000 on your Databricks." is really strange as well.
What is "your Databricks"?

**Review by FabienLelaquais — APPROVED** [review]
👍

**Review by jrobinAV — APPROVED** [review]
We should ask the CS team/tech writer to improve the diagrams. Meanwhile that s good to be merge for me.

**Pull request merged**